targetselector/running_target_selector.go

```diff
@@ -64,22 +75,25 @@ func (r runningTargetSelector) GetMatchingPodsOverTotalPods(c client.Client, ins
 			}
 		}
 
-		// if the disruption is applied on init, we only target pending pods with a running
+		// if the disruption is applied on init, we only target pending pods with a running (or terminated)
 		// chaos handler init container
 		// otherwise, we only target running pods
 		if instance.Spec.OnInit {
 			hasChaosHandler := false
 
 			// search for a potential running chaos handler init container
 			for _, initContainerStatus := range pod.Status.InitContainerStatuses {
```

Comment:
With dynamicTargeting, the targeted pod(s) from an on init disruption is removed because the status changes from Pending to Running and the chaos-handler init container is then Terminated.